src/include/sof/math/fir_config.h

```diff
@@ -21,41 +21,42 @@
 #endif
 #endif
 
-/* If next defines are set to 1 the EQ is configured automatically. Setting
- * to zero temporarily is useful is for testing needs.
- * Setting EQ_FIR_AUTOARCH to 0 allows to manually set the code variant.
+/* Define SOFM_FIR_FORCEARCH 0/2/3 in build command line or temporarily in
+ * this file to override the default auto detection.
  */
-#define FIR_AUTOARCH    1
-
-/* Force manually some code variant when EQ_FIR_AUTODSP is set to zero. These
- * are useful in code debugging.
- */
-#if FIR_AUTOARCH == 0
-#define FIR_GENERIC	0
-#define FIR_HIFIEP	0
-#define FIR_HIFI3	1
-#endif
-
-/* Select optimized code variant when xt-xcc compiler is used */
-#if FIR_AUTOARCH == 1
-#if defined __XCC__
-#include <xtensa/config/core-isa.h>
-#define FIR_GENERIC	0
-#if XCHAL_HAVE_HIFI2EP == 1
-#define FIR_HIFIEP	1
-#define FIR_HIFI3	0
-#elif XCHAL_HAVE_HIFI3 == 1 || XCHAL_HAVE_HIFI4 == 1
-#define FIR_HIFI3	1
-#define FIR_HIFIEP	0
+#ifdef SOFM_FIR_FORCEARCH
```

Comment:
where and how to define SOFM_FIR_FORCEARCH and SOF_IIR_DF2T_FORCEARCH? feel this patch make code more complex than before change, not sure whether we can have a more straightforward code clean.

Reply:
> where and how to define SOFM_FIR_FORCEARCH and SOF_IIR_DF2T_FORCEARCH?

The answer is literally in the first line of the patch.

> feel this patch make code more complex than before change, 

That's a bold statement considering you didn't read the first line.